src/pykeen/triples/triples_factory.py

```diff
@@ -1396,3 +1533,464 @@ def get_mapped_triples(
+        )
+        return self.label_quadruples(self.mapped_quadruples)
+
+    def label_quadruples(
+        self,
+        quadruples: MappedQuadruples,
+        unknown_entity_label: str = "[UNKNOWN]",
+        unknown_relation_label: Optional[str] = None,
+        unknown_timestamp_label: Optional[str] = None,
+    ) -> LabeledQuadruples:
+        """
+        Convert ID-based quadruples to label-based ones.
+
+        :param quadruples:
+            The ID-based quadruples.
+        :param unknown_entity_label:
+            The label to use for unknown entity IDs.
+        :param unknown_relation_label:
+            The label to use for unknown relation IDs.
+        :param unknown_timestamp_label:
+            The label to use for unknown timestamp IDs.
+        :return:
+            The same triples, but labeled.
+        """
+        if len(quadruples) == 0:
+            return np.empty(shape=(0, 4), dtype=str)
+        if unknown_relation_label is None:
+            unknown_relation_label = unknown_entity_label
+        if unknown_timestamp_label is None:
+            unknown_timestamp_label = unknown_entity_label
+    
+        return np.stack(
+            [
+                labeling.label(ids=column, unknown_label=unknown_label)
+                for (labeling, unknown_label), column in zip(
+                    [
+                        (self.entity_labeling, unknown_entity_label),
+                        (self.relation_labeling, unknown_relation_label),
+                        (self.entity_labeling, unknown_entity_label),
+                        (self.timestamp_labeling, unknown_timestamp_label),
+                    ],
+                    quadruples.t().numpy(),
+                )
+            ],
+            axis=1,
+        )
+
+    def timestamps_to_ids(
+        self, timestamps: Union[Collection[int], Collection[str]]
+    ) -> Collection[int]:  # noqa: D102
+        return _ensure_ids(
+            labels_or_ids=timestamps, label_to_id=self.timestamp_labeling.label_to_id
+        )
+
+    def timestamp_word_cloud(self, top: Optional[int] = None):
+        """Make a word cloud based on the frequency of occurrence of each timestamp in a Jupyter notebook.
+
+        :param top: The number of top timestamps to show. Defaults to 100.
+        :returns: A world cloud object for a Jupyter notebook
+
+        .. warning::
+
+            This function requires the ``wordcloud`` package. Use ``pip install pykeen[wordcloud]`` to install it.
+        """
+        return self._word_cloud(
+            ids=self.mapped_quadruples[:, 3],
+            id_to_label=self.timestamp_labeling.id_to_label,
+            top=top or 100,
+        )
+
+    def tensor_to_df(
+        self,
+        tensor: torch.LongTensor,
+        **kwargs: Union[torch.Tensor, np.ndarray, Sequence],
+    ) -> pd.DataFrame:  # noqa: D102
+        data = super().tensor_to_df(tensor=tensor, **kwargs)
```

Comment:
Does this work? Shouldn't the tensor here be of shape `(*, 4)`, but super accepting only `(*, 3)`?